selfdrive/car/subaru/carstate.py

```diff
@@ -51,18 +51,30 @@ def update(self, cp, cp_cam):
     ret.cruiseState.enabled = cp.vl["CruiseControl"]['Cruise_Activated'] != 0
     ret.cruiseState.available = cp.vl["CruiseControl"]['Cruise_On'] != 0
     ret.cruiseState.speed = cp_cam.vl["ES_DashStatus"]['Cruise_Set_Speed'] * CV.KPH_TO_MS
-    # EDM Impreza: 1 = mph, UDM Forester: 7 = mph
-    if cp.vl["Dash_State"]['Units'] in [1, 7]:
-      ret.cruiseState.speed *= CV.MPH_TO_KPH
+
+    # UDM Legacy: mph = 0
+    if self.car_fingerprint == CAR.LEGACY_PREGLOBAL:
+      if (cp.vl["Dash_State"]['Units'] == 0):
+        ret.cruiseState.speed *= CV.MPH_TO_KPH
+    # EDM Global: mph = 1, 2
+    else:
+      if cp.vl["Dash_State"]['Units'] in [1, 2]:
+        ret.cruiseState.speed *= CV.MPH_TO_KPH
 
     ret.seatbeltUnlatched = cp.vl["Dashlights"]['SEATBELT_FL'] == 1
     ret.doorOpen = any([cp.vl["BodyInfo"]['DOOR_OPEN_RR'],
                         cp.vl["BodyInfo"]['DOOR_OPEN_RL'],
                         cp.vl["BodyInfo"]['DOOR_OPEN_FR'],
                         cp.vl["BodyInfo"]['DOOR_OPEN_FL']])
 
-    self.es_distance_msg = copy.copy(cp_cam.vl["ES_Distance"])
-    self.es_lkas_msg = copy.copy(cp_cam.vl["ES_LKAS_State"])
+    if self.car_fingerprint not in PREGLOBAL_CARS:
```

Comment:
All the nots make it harder to read. I think it's fine if they all start with the preglobals

Reply:
I'll revert back to preglobal first